src/engine/source/indexerconnector/test/component/indexerConnector_test.cpp

```diff
@@ -717,3 +717,85 @@ TEST_F(IndexerConnectorTest, PublishDatePlaceholder)
     ASSERT_NO_THROW(indexerConnector.publish(publishData.dump()));
     ASSERT_NO_THROW(waitUntil([&callbackCalled]() { return callbackCalled; }, MAX_INDEXER_PUBLISH_TIME_MS));
 }
+
+/**
+ * @brief Test the connection and posterior data publication into a server. The published data is checked against the
+ * expected one, with and without an index name in the request.
+ *
+ */
+TEST_F(IndexerConnectorTest, PublishDatePlaceholderIndexNameInRequest)
+{
+    // We create an index with the current date as part of the name.
+    std::string baseIndexName = std::string(INDEXER_NAME) + "_$(date)";
+    // We'll create two indexes name, one will have the date replaced and the other will have a custom string.
+    std::string indexName1 = baseIndexName;
+    base::utils::string::replaceAll(indexName1, "$(date)", base::utils::time::getCurrentDate("."));
+    std::string indexName2 = baseIndexName;
+    base::utils::string::replaceAll(indexName2, "$(date)", "custom_string");
```

Comment:
Tests should validate that the `$(date)` placeholder replacement format is applied correctly.

Reply:
That is verified inside the lambda.
But the test will change now that I'll remove part of the feature

Reply:
The new approach doesn't contain test because:
- The basic feature is already tested in previous tests
- To verify the change in the date, there are two options (and I'm not sure about any): I can templetize the `IndexerConnector` class to mock the `std::time` call or I can change the host date during the test

Reply:
Thanks!